File: frogoui/src/main/java/com/frogobox/ui/fireworks/initializers/RotationSpeedInitializer.java
package com.frogobox.ui.fireworks.initializers;

import com.frogobox.ui.fireworks.Particle;

import java.util.Random;

public class RotationSpeedInitializer implements ParticleInitializer {

    private final float mMinRotationSpeed;
    private final float mMaxRotationSpeed;

    public RotationSpeedInitializer(float minRotationSpeed, float maxRotationSpeed) {
        mMinRotationSpeed = minRotationSpeed;
        mMaxRotationSpeed = maxRotationSpeed;
    }

    @Override
    public void initParticle(Particle p, Random r) {
        float rotationSpeed = r.nextFloat() * (mMaxRotationSpeed - mMinRotationSpeed) + mMinRotationSpeed;
        p.mRotationSpeed = rotationSpeed;
    }

}

File: core-ui-android/src/main/java/com/frogobox/ui/fireworks/initializers/RotationSpeedInitializer.kt
package com.frogobox.ui.fireworks.initializers

import com.frogobox.ui.fireworks.Particle
import java.util.*

class RotationSpeedInitializer(
    private val mMinRotationSpeed: Float,
    private val mMaxRotationSpeed: Float
) : ParticleInitializer {

    override fun initParticle(p: Particle?, r: Random?) {
        val rotationSpeed =
            r!!.nextFloat() * (mMaxRotationSpeed - mMinRotationSpeed) + mMinRotationSpeed
        p!!.mRotationSpeed = rotationSpeed
    }

}